internal/controllers/reconciliation/controller.go

```diff
@@ -123,7 +123,7 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("getting resource slice: %w", err)
 	}
-	if len(slice.Status.Resources) > req.Manifest.Index {
+	if len(slice.Status.Resources) >= req.Manifest.Index-1 {
```

Comment:
I think the original logic was fine? if length == 5 and index == 6 this passes and it shouldn't

Reply:
lol you're right. That's what I get for coding during meetings 😬